lib/exchange/export_compilation.cpp

```diff
@@ -0,0 +1,112 @@
+//*                             _    *
+//*   _____  ___ __   ___  _ __| |_  *
+//*  / _ \ \/ / '_ \ / _ \| '__| __| *
+//* |  __/>  <| |_) | (_) | |  | |_  *
+//*  \___/_/\_\ .__/ \___/|_|   \__| *
+//*           |_|                    *
+//*                            _ _       _   _              *
+//*   ___ ___  _ __ ___  _ __ (_) | __ _| |_(_) ___  _ __   *
+//*  / __/ _ \| '_ ` _ \| '_ \| | |/ _` | __| |/ _ \| '_ \  *
+//* | (_| (_) | | | | | | |_) | | | (_| | |_| | (_) | | | | *
+//*  \___\___/|_| |_| |_| .__/|_|_|\__,_|\__|_|\___/|_| |_| *
+//*                     |_|                                 *
+//===- lib/exchange/export_compilation.cpp --------------------------------===//
+//
+// Part of the pstore project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://github.com/SNSystems/pstore/blob/master/LICENSE.txt for license
+// information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+/// \file export_compilation.cpp
+/// \brief  Functions for exporting compilations and the compilation index.
+#include "pstore/exchange/export_compilation.hpp"
+
+#include "pstore/exchange/export_ostream.hpp"
+
+using pstore::exchange::export_ns::ostream_base;
+
+namespace pstore {
+    namespace repo {
+
+#define X(a)                                                                                       \
+    case repo::linkage::a: return os << #a;
+        ostream_base & operator<< (ostream_base & os, linkage const l);
+        ostream_base & operator<< (ostream_base & os, linkage const l) {
+            switch (l) { PSTORE_REPO_LINKAGES }
+            return os << "unknown";
+        }
+#undef X
+
+        ostream_base & operator<< (ostream_base & os, visibility const v);
```

Comment:
Next line is the definition of `operator<<`. Why need this line?

Reply:
It avoids a compiler warning that an external function doesn’t have a declaration.

Reply:
As above.